Decred Wallet/Features/Transactions/TransactionTableViewCell.swift

```diff
@@ -12,84 +12,95 @@ import Dcrlibwallet
+    @IBOutlet weak var voteInfoLabel: Label!
     @IBOutlet weak var status: UILabel!
-   // @IBOutlet weak var dateT: UILabel!
+    @IBOutlet weak var secondaryStatusLabel: UILabel!
     @IBOutlet weak var statusIcon: UIImageView!
-    
+
     var count = 0
-    
+
     override func awakeFromNib() {}
-    
+
     override class func height() -> CGFloat {
         return 56
     }
-    
+
     func setData(_ data: Any?) {
         if let transaction = data as? Transaction {
             var confirmations: Int32 = 0
             if transaction.blockHeight != -1 {
                 confirmations = WalletLoader.shared.firstWallet!.getBestBlock() - Int32(transaction.blockHeight) + 1
             }
-            let Date2 = NSDate.init(timeIntervalSince1970: TimeInterval(transaction.timestamp) )
-                      let dateformater = DateFormatter()
-                      dateformater.locale = Locale(identifier: "en_US_POSIX")
-                      dateformater.dateFormat = "MMM dd"
-                      dateformater.string(from: Date2 as Date)
-                      
-                     // self.dateT.text = dateformater.string(from: Date2 as Date)
-            
+
+            let txDate = NSDate.init(timeIntervalSince1970: TimeInterval(transaction.timestamp))
+            let dateformater = DateFormatter()
+            dateformater.locale = Locale(identifier: "en_US_POSIX")
+            dateformater.dateFormat = "MMM dd"
+            dateformater.string(from: txDate as Date)
+            let ageInDays = (txDate as Date).daysFromNow
+            var dateString = dateformater.string(from: txDate as Date)
+
+            if ageInDays == 0 {
+                dateString = LocalizedStrings.today
+            } else if ageInDays == -1 {
+                dateString = LocalizedStrings.yesterday
+            }
+
             let isConfirmed = Settings.spendUnconfirmed || confirmations > 1
-            self.status.text = isConfirmed ? dateformater.string(from: Date2 as Date) : LocalizedStrings.pending
-            self.status.textColor = isConfirmed ? UIColor(hex:"#596d81") : UIColor(hex:"#8997a5")
-            self.statusIcon.image = isConfirmed ? UIImage(named: "confirmed") : UIImage(named: "pending")
-            
-          
+            self.status.text = isConfirmed ? dateString : LocalizedStrings.pending
+            self.status.textColor = isConfirmed ? UIColor.appColors.bluishGray : UIColor.appColors.lightBluishGray
+            self.statusIcon.image = isConfirmed ? UIImage(named: "ic_confirmed") : UIImage(named: "ic_pending")
+
             let requireConfirmation = Settings.spendUnconfirmed ? 0 : 2
-            
+
+            self.secondaryDataLabel.isHidden = transaction.type == DcrlibwalletTxTypeRegular
+            self.secondaryStatusLabel.isHidden = transaction.type == DcrlibwalletTxTypeRegular
+            self.voteInfoLabel.isHidden = true //TODO: set value
+
             if transaction.type == DcrlibwalletTxTypeRegular {
```

Comment:
Break this mighty nested if blocks into functions and refactor for readability and maintainabilty:
- `displayRegularTxInfo(_ transaction: Transaction)`
- `displayTicketPurchaseInfo(_ transaction: Transaction)`
- `displayVoteTxInfo(_ transaction: Transaction)`